File: PHP/Idades.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

//echo "The result is "+calcAverage([34, 56, 44, 23]);

class Idades {

    /**
     * Calculate and return the average of ages.
     */
    public function calcAverage( $ages = [] ) {
        $result = 0;

        if ( !empty( $ages ) ) {
            $result = array_sum( $ages ) / count( $ages );
        }

        return $result;
    }
}

File: GO/Idades.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

/**
 * Calculate and return the average of ages.
 */
func calcAverage(ages []float64) float64 {
	var result float64

	for _, age := range ages {
		result += age
	}

	return result / float64(len(ages))
}
